squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/ErrorLoggingPlugin.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the Apache 2.0 License.
+ * See the accompanying LICENSE file for terms.
+ */
+package com.yahoo.squidb.processor.plugins.defaults;
+
+import com.yahoo.aptutils.model.DeclaredTypeName;
+import com.yahoo.aptutils.utils.AptUtils;
+import com.yahoo.aptutils.writer.JavaFileWriter;
+import com.yahoo.aptutils.writer.expressions.Expression;
+import com.yahoo.aptutils.writer.expressions.Expressions;
+import com.yahoo.aptutils.writer.parameters.TypeDeclarationParameters;
+import com.yahoo.squidb.processor.data.ErrorInfo;
+import com.yahoo.squidb.processor.data.ModelSpec;
+import com.yahoo.squidb.processor.plugins.Plugin;
+import com.yahoo.squidb.processor.plugins.PluginEnvironment;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Modifier;
+
+/**
+ * A {@link Plugin} that controls writing any errors logged using {@link ModelSpec#logError(String, Element)} to the
+ * generated file using the {@link com.yahoo.squidb.annotations.ModelGenErrors} annotation, to be subsequently
+ * processed by the {@link com.yahoo.squidb.processor.ErrorLoggingProcessor}. It is enabled by default.
+ * It can be disabled by passing the {@link PluginEnvironment#OPTIONS_DISABLE_ERROR_LOGGING} option in the
+ * squidbOptions key.
+ */
+public class ErrorLoggingPlugin extends Plugin {
+
+    private static final String MODEL_GEN_ERRORS_CLASS = "com.yahoo.squidb.annotations.ModelGenErrors";
+    private static final DeclaredTypeName MODEL_GEN_ERRORS = new DeclaredTypeName(MODEL_GEN_ERRORS_CLASS);
+    private static final DeclaredTypeName MODEL_GEN_ERROR_INNER =
+            new DeclaredTypeName(MODEL_GEN_ERRORS_CLASS, "ModelGenError");
+
+    public ErrorLoggingPlugin(ModelSpec<?> modelSpec, PluginEnvironment pluginEnv) {
+        super(modelSpec, pluginEnv);
+    }
+
+    @Override
+    public void addRequiredImports(Set<DeclaredTypeName> imports) {
+        if (modelSpec.getLoggedErrors().size() > 0) {
+            imports.add(MODEL_GEN_ERRORS);
+            imports.add(MODEL_GEN_ERROR_INNER);
+        }
+    }
+
+    @Override
+    public void emitAdditionalJava(JavaFileWriter writer) throws IOException {
```

Comment:
Interesting. I expected the annotations would go on the class itself, but given how the writer works, this makes sense (and nobody should have to look at it in source anyway since they're printed by the build tools).